packages/devtools/lib/src/timeline/frames_bar_chart.dart

```diff
@@ -97,28 +97,78 @@ class PlotlyDivGraph extends CoreElement {
     }
   }
 
+  Selection currentSelection = null;
+
   void _plotlyClick(DataEvent data) {
     final int xPosition = data.points[0].x;
-    if (_frames.containsKey(xPosition)) {
-      final TimelineFrame timelineFrame = _frames[xPosition];
-      framesBarChart.setSelected(timelineFrame);
+
+    final List<int> pointNumbers = [];
+    final List<int> xValues = [];
+    final List<num> yValues = [];
+    final List<int> traces = [];
+
+    for (Point pt in data.points) {
+      // Don't allow selecting an already selected bar.
+      if (pt.curveNumber != FramesBarPlotly.gpuSelectTraceIndex &&
+          pt.curveNumber != FramesBarPlotly.cpuSelectTraceIndex) {
+        pointNumbers.add(pt.pointNumber);
+        traces.add(pt.curveNumber);
+        xValues.add(pt.x);
+        yValues.add(pt.y);
+      }
+    }
+
+    // Create selection once.
+    currentSelection ??= Selection(frameGraph, element);
+
+    // If this bar isn't currently selected then select the bar clicked.
+    if (xValues.length == 2 &&
+        !currentSelection.isSelected(xValues)) {
+      currentSelection.select(pointNumbers, traces, xValues, yValues);
+
+      if (_frames.containsKey(xPosition)) {
+        final TimelineFrame timelineFrame = _frames[xPosition];
+        framesBarChart.setSelected(timelineFrame);
+      }
+    } else {
+      _plotlyHover(data);
     }
   }
 
   void _plotlyHover(DataEvent data) {
     final List<HoverFX> hoverDisplay = [];
 
+    // Hovering over a selected bar?
+    bool selectionTrace = false;
+
     for (Point pt in data.points) {
       final int ptNumber = pt.pointNumber;
       final int x = pt.data.x[ptNumber];
       // Only display the hover if its not the first data point for each trace
       // (curveNumber). Works around first bar in a trace color not rendered.
       if (x != FramesBarPlotly.xCoordNotUsed) {
+        selectionTrace |=
+            (pt.curveNumber == FramesBarPlotly.gpuSelectTraceIndex) ||
+                (pt.curveNumber == FramesBarPlotly.cpuSelectTraceIndex);
         hoverDisplay.add(
             HoverFX(curveNumber: pt.curveNumber, pointNumber: pt.pointNumber));
       }
     }
 
+    if (selectionTrace) {
+      // Hide the hover of the gpu good/jank & cpu good/jank trace of the
+      // selected bar.
```

Comment:
is there a reason we don't still want to show the duration values on hover?

Reply:
There was before I was removing the point from a trace.  As I cleaned up the SelectTrace in plotlyClick function I removed this too.